source/server/http/admin.cc

```diff
@@ -284,6 +285,17 @@ Http::Code AdminImpl::handlerQuitQuitQuit(const std::string&, Buffer::Instance&
   return Http::Code::OK;
 }
 
+Http::Code AdminImpl::handlerListenerAddresses(const std::string&, Buffer::Instance& response) {
+  std::list<std::string> listeners;
+  int listener_index = 0;
+  while (server_.getListenSocketByIndex(listener_index) != nullptr) {
```

Comment:
Can write this as `while (auto listen_socket = server_.getListenSocketByIndex(listener_index) ) { ...`